### DISCUSSION TITLE
docker container indexer certificate issue ?

### DISCUSSION THREAD

**Discussion by guedz** [wazuh/wazuh-docker#1611] [General]
Hello,

before declaring it as an issue, I'd like to be sure that it's not an error on my part about deploying wazuh via docker :

my concern is the crash at startup of the indexer container on the error message : 

```
Caused by: org.opensearch.OpenSearchSecurityException: Error while initializing transport SSL layer from PEM: java.security.AccessControlException: access denied ("java.io.FilePermission" "/etc/wazuh-indexer/certs/indexer.pem" "read")
```

I've tested on Windows (to avoid some of the rights issues that can occur on Linux) or Linux with the same result 

I think I'm following your deployment procedure:

- clone git project
- create certificate from image source : wazuh/wazuh-certs-generator:0.0.2
- use single-node docker-compose.yml
- use docker compose up -d command

From wazuh-certs-generator i have the following files :

admin-key.pem
admin.pem
root-ca-manager.key
root-ca-manager.pem
root-ca.key
root-ca.pem
wazuh.dashboard-key.pem
wazuh.dashboard.pem
wazuh.indexer-key.pem
wazuh.indexer.pem
wazuh.manager-key.pem
wazuh.manager.pem

I've had a quick look at the entrypoint and other scripts in the git repository and I don't understand how you get this error message.
Where does this reference to /etc/wazuh-indexer/certs/ come from?

Thanks for your help
Best regards
Guedz

**Comment by guedz** [comment]
from a test with docker desktop :

![image](https://github.com/user-attachments/assets/78777ff8-295f-4f11-8237-e0446444a198)

No /etc/wazuh-indexer/ folder present on the container.

environnement variable issue ?

**Comment by sierja** [comment]
Same problem with 4.11.2.

I'm seeing a number of issues filed that probably all have the same, or similar, root cause from the last few months.

@c-bordon When was the last time someone tried spinning up a single-node setup based on the [official instructions](https://documentation.wazuh.com/current/deployment-options/docker/wazuh-container.html#single-node-deployment)? Are we sure the docker-compose.yml is still current and valid?
